Add Android build target

jasuarez opened the pull request:
This PR enables to have an "android" build target that allows to compile the DRI drivers for Android.

It's based on Rob Herring work for Linaro (https://ci.linaro.org/job/robher-aosp).

This fixes issue #11.

jasuarez committed:
mesa: add android build target

This new build targets compiles Mesa drivers for Linaro's ARM64 and
X86_64 AOSP.

tanty commented on Rockerfile.mesa:
```diff
@@ -29,31 +31,73 @@ FROM {{ $image }}:llvm-{{ .LLVM }}
+     {{ if eq .BUILD "windows" }}mingw-w64{{ end }}     \
+     {{ if eq .BUILD "android" }}                       \
+       wget openjdk-8-jdk python sudo make gcc gettext  \
+       python-mako bison libc6-dev xz-utils             \
```
I suppose it doesn't matter much but it feels like wget, python, sudo, make gcc, gettext, python-mako, bison, libc6-dev and xz-utils should be already installed, aren't they?

jasuarez replied:
Not really. For Android we are using as base system Ubuntu Xenial, not our base. Reason is that base contains a lot of dependencies that are not useful at lot for Android, and thus we can reduce the final size.

tanty commented on Rockerfile.mesa:
```diff
@@ -29,31 +31,73 @@ FROM {{ $image }}:llvm-{{ .LLVM }}
+  && mv repo /home/local/.bin
+
+ENV PATH=/usr/lib/ccache:/home/local/.bin/:$PATH
+ENV USE_CCACHE=1
```
I think we should move to the same point where the MOUNT for ccache is done:

ENV PATH=/usr/lib/ccache:$PATH
ENV USE_CCACHE=1

and leave only the /home/local/.bin and the /usr/lib/llvm-{{ $llvm_version }}/bin to the different branches of the "if"

Also, notice that the USE_CCACHE=1 is only exported in the Android branch (AFAIS)

jasuarez replied:
Sure. I'll move it.

OTOH, yeah, USE_CCACHE is an envvar that Android uses to enable/disable ccache. By default, in Android ccache is disabled, thus we need to explicitly enable here.

jasuarez replied:
Thinking more about it, I think I'll leave as it is. The thing is that I want to ensure that ccache is always the first in the PATH. If I set it at the beginning and later set local/.bin or llvm-bin (depending if it is Android or no) I don't know how to put it after the ccache path.

I'll push as it is, and if later we find a way to fix it, we will do.

tanty reviewed (approved):
Other than those 2 nitpicks, this is :+1: :smiley:

jasuarez committed:
Base Android image directly over Ubuntu Xenial

For this build, we do not need any of the requirements installed in
base, as Android already fetches its own dependencies.

This helps to reduce the total size of the image.

jasuarez commented:
This fixes issue #23
